lib/commands/execute.js

```diff
@@ -44,29 +10,24 @@ const EXTENSION_COMMANDS_MAPPING = {
  * @returns {Promise<any>}
  */
 export async function execute (script, args) {
-  if (script.match(/^macos:/)) {
-    this.log.info(`Executing extension command '${script}'`);
-    script = script.replace(/^macos:/, '').trim();
-    return await this.executeMacosCommand(script, _.isArray(args) ? args[0] : args);
-  }
-  throw new errors.NotImplementedError();
+  this.log.info(`Executing extension command '${script}'`);
+  const formattedScript = String(script).trim().replace(/^macos:\s*/, `${EXECUTE_SCRIPT_PREFIX} `);
+  const preprocessedArgs = preprocessExecuteMethodArgs(args);
+  return await this.executeMethod(formattedScript, [preprocessedArgs]);
 };
 
 /**
+ * Massages the arguments going into an execute method.
  *
- * @this {Mac2Driver}
- * @param {string} command
- * @param {import('@appium/types').StringRecord} [opts={}]
- * @returns {Promise<any>}
+ * @param {ExecuteMethodArgs} [args]
+ * @returns {StringRecord}
  */
-export async function executeMacosCommand (command, opts = {}) {
-  if (!_.has(EXTENSION_COMMANDS_MAPPING, command)) {
-    throw new errors.UnknownCommandError(`Unknown extension command "${command}". ` +
-      `Only ${_.keys(EXTENSION_COMMANDS_MAPPING)} commands are supported.`);
-  }
-  return await this[/** @type {string} */ (EXTENSION_COMMANDS_MAPPING[command])](opts);
-};
+function preprocessExecuteMethodArgs(args) {
+  return /** @type {StringRecord} */ ((_.isArray(args) ? _.first(args) : args) ?? {});
```

Comment:
we should also unwrap elements properly as part of the preprocessing. See  https://github.com/appium/appium-android-driver/blob/b98e6fd3cabec0cf16f01018b0d0e43802bef58a/lib/commands/execute.js#L51